src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -144,14 +144,19 @@ public static async Task<string> DownloadFileAsync(string url, long? maxDownload
                 throw new DownloadSizeExceededException(maxDownloadSize.Value);
             }
 
-            if (!File.Exists(targetFile) || new FileInfo(targetFile).Length != downloadSize)
+            if (File.Exists(targetFile))
             {
-                File.Delete(targetFile);
-                using var targetFileStream = File.OpenWrite(targetFile);
-                var contentStream = await response.Content.ReadAsStreamAsync();
-                await contentStream.CopyToAsync(targetFileStream);
+                string fileName = Path.GetFileNameWithoutExtension(targetFile);
+                string fileExt = Path.GetExtension(targetFile);
+                DirectoryInfo dir = new DirectoryInfo(InstallerDownloadPath);
+                FileInfo[] existingFiles = dir.GetFiles(Path.ChangeExtension(fileName + " *", fileExt));
+                targetFile = Path.Combine(InstallerDownloadPath, fileName + " (" + (existingFiles.Length + 1).ToString() + ")" + fileExt);
```

Comment:
Can you separate this out into it's own function so it's easier to read and can be used elsewhere if needed? Maybe like `GetNumericFilename()`